src/azure-cli-core/azure/cli/core/local_context.py

```diff
@@ -7,47 +7,90 @@
 import stat
 import shutil
 import configparser
+import enum
 
 from knack.log import get_logger
 from knack.config import _ConfigFile
-from knack.util import ensure_dir, CLIError
+from knack.util import ensure_dir
 
-SET = 'SET'   # action for a parameter in local context, SET means its value will be saved to local context
-GET = 'GET'   # action for a parameter in local context, GET means will read value from local context for this parameter
 ALL = 'ALL'   # effective level of local context, ALL means all commands can share this parameter value
+LOCAL_CONTEXT_FILE = '.local_context_{}'
+LOCAL_CONTEXT_CONFIG_SECTION = 'local_context'
+LOCAL_CONTEXT_NOTICE = '; This file is used to store local context data.\n'\
+                       '; DO NOT modify it manually unless you know it well.\n'
 logger = get_logger(__name__)
 
 
+class LocalContextAction(enum.Enum):
+    SET = 1   # action for a parameter in local context, SET means its value will be saved to local context
+    GET = 2   # action for a parameter in local context, GET means will read value from local context for this parameter
+
+
+def _get_current_username(cli_ctx):
+    from azure.cli.core._profile import Profile
+    profile = Profile(cli_ctx=cli_ctx)
+    username = None
+    try:
+        username = profile.get_current_account_user()
+    except Exception:  # pylint: disable=broad-except
+        pass
+    return username
+
+
 class AzCLILocalContext(object):
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=attribute-defined-outside-init
+
+    def __init__(self, cli_ctx):
+        self.cli_ctx = cli_ctx
+        self.config = cli_ctx.config
+        self.dir_name = os.path.basename(self.config.config_dir)
+        self.initialize()
+
+    def initialize(self):
+        self.username = _get_current_username(self.cli_ctx)
+        if self.username is not None:
+            self.file_name = LOCAL_CONTEXT_FILE.format(self.username)
+            self.is_on = self.config.getboolean(LOCAL_CONTEXT_CONFIG_SECTION, self.username, False)
+        else:
+            self.is_on = False
 
-    def __init__(self, dir_name=None, file_name=None):
-        self.dir_name = dir_name
-        self.file_name = file_name
-        self._local_context_file = None
-        self._load_local_context_file()
+        self.current_dir = None
+        try:
+            self.current_dir = os.getcwd()
+        except FileNotFoundError:
+            logger.debug('The working directory has been deleted or recreated. Local context is ignored.')
+
+        self.local_context_file = None
+        if self.is_on and self.current_dir:
```

Comment:
move to _load_local_context_file()?

Reply:
There are some logic updates here. `_load_local_context_file` update to `_load_local_context_files`, it may also be called if local context is off (show/delete command when local context is off). Also update `self.local_context_file` to `self._local_context_file`, it is loaded only once(when local context is on) and only used for `get`/`set` for local context parameter value.